KIPs/kip-81.md

```diff
@@ -0,0 +1,122 @@
+- Expired: Executor has not queued or executed within respective timeouts.
+
+
+![proposal state diagram](../assets/kip-81/proposal_state_diagram.png)
+
+For a proposal to pass, two quorum conditions must be satisfied. The first condition is the turnout. Either the number of voters exceeds 1/3 of all voters, or the sum of casted votes exceeds 1/3 of all eligible votes. The second condition is supported. The yes votes must exceed the half of casted votes.
+The amount of votes are calculated from the amount of stakes KLAY. GC members stake their KLAY by locking in a special smart contract called CnStakingContract. Each member must deploy at least one CnStakingContract and deposit KLAY there. The official addresses of CnStakingContract are managed in another special smart contract called AddressBook.
+Each GC member can appoint a voter account that will cast votes on behalf of the KLAY in its CnStakingContracts. The voter account is necessary because CnStakingContract itself cannot send transactions to cast votes.
+
+![gc voting structure](../assets/kip-81/gc_voting_structure.png)
+  
+Staked balances and voter accounts are tracked in a new special smart contract named StakingTracker. StakingTracker stores the voter list and their voting powers for each voting period. CnStakingContract must notify StakingTracker to recognize any balance changes. Below is an example of StakingTracker contract interface.
+```solidity
+interface IStakingTracker {
+
+  struct Tracker {
+    // this struct is updated only if trackStart <= block.number <= trackEnd.
+    uint64 trackStart;
+    uint64 trackEnd;
+
+    // List of eligible nodes and their staking contracts.
+    // Determined at createTracker() and won’t change.
+    address[] nodeIds;
+    mapping(address => address) stakingToNodeId;
+
+
+    // Balances and voting powers of each nodes.
+    // Updated by notifyStake() and notifyWithdraw().
+    mapping(address => uint256) nodeBalances;
+    mapping(address => uint256) votingPowers;
+    uint256 totalVotes;
+
+
+    // Appointed voters list
+    // Updated by appointVoter().
+    mapping(address => address) nodeIdToVoter;
+    mapping(address => address) voterToNodeId;
+  }
+
+  // Called by Voting at proposal submission, to finalize voter list.
+  function createTracker(uint256 trackEnd) public returns (uint trackerId);
+
+
+  // Called by CnStakingContract whenever its balance changes.
+  function notifyStake(uint256 amount) public;
+  function notifyWithdraw(uint256 amount) public;
+
+  /// Called by Governance admin or CnStaking admins to change voter account.
+  function appointVoter(address nodeId, address voter) public;
+
+  /// Called by Voting to calculate voting outcome.
+  function getVotingPower(uint trackerId, address voter)
+    public view returns(uint256);
+  function quorums(uint trackerId)
+    public view returns(uint quorumCount, uint quorumPower); 
+}
+```
```

Comment:
It also could be good to provide an example implementation of StakingTracer for better understanding of the interfaces.

Reply:
I will add a simplfied implementation.

Reply:
@kjhman21 Refer to sample implementation in "StakingTracker contract" section.